### PR TITLE
feat(alerts): Add insights metrics option and alert type

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds Insights Metric as an Alert type in the UI, protected behind the `insights-alerts` feature flag.
image

### CODE REVIEW

**Comment by edwardgou-sentry** [comment]
note: The alert rule builder doesn't fully work with Insight Metrics yet. This PR just includes the option and some types + constants. There will be a separate PR to update the builder to fully support Insights

**Review by nhsiehgit — APPROVED** [review]
Just need a :rubber_stamp:? 🤷 👍

**Comment by edwardgou-sentry** [comment]
Thanks for the review @nhsiehgit ! This PR relies on some changes from #76566 which is still open. I based this PR on that branch to minimize the code diff. Let me know if that changes anything for you! I plan to merge this in to `master` after #76566 is reviewed and merged

**Review by priscilawebdev — APPROVED** [review]
code looks good

File: static/app/views/alerts/wizard/panelContent.tsx
Comment (edwardgou-sentry) [reply static/app/views/alerts/wizard/panelContent.tsx:177]: Yes Insights metrics are the charts rendered in the Insights modules! We're planning to add `Create Alert` buttons to those charts in the Insights Modules that will take users directly to the alert builder page. I'll make sure to update docs around this as well!

**Pull request merged**
